src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Requests/DataSetQueryRequest.cs

```diff
@@ -44,4 +45,38 @@ public record DataSetQueryRequest
     /// </summary>
     [DefaultValue(1000)]
     public int PageSize { get; init; } = 1000;
+
+    public class Validator : AbstractValidator<DataSetQueryRequest>
+    {
+        public Validator()
+        {
+            RuleFor(q => q.Indicators)
+                .NotEmpty();
+            RuleForEach(q => q.Indicators)
+                .NotEmpty()
+                .MaximumLength(40);
+
+            RuleFor(q => q.Criteria)
+                .SetInheritanceValidator(v =>
+                {
+                    v.Add(new DataSetQueryCriteriaAnd.Validator());
+                    v.Add(new DataSetQueryCriteriaOr.Validator());
+                    v.Add(new DataSetQueryCriteriaNot.Validator());
+                    v.Add(new DataSetQueryCriteriaFacets.Validator());
+                });
+
+            When(q => q.Sorts is not null, () =>
+            {
+                RuleFor(q => q.Sorts)
+                    .NotEmpty();
+                RuleForEach(q => q.Sorts)
+                    .SetValidator(new DataSetQuerySort.Validator());
```

Comment:
Do we need to add a `.NotNull()` here for each item in the `Sorts` list?

Reply:
Good point, I've added this in